src/serac/physics/base_physics.cpp

```diff
@@ -356,21 +356,34 @@ FiniteElementState BasePhysics::loadCheckpointedState(const std::string& state_n
   return checkpoint_states_.at(state_name)[static_cast<size_t>(cycle)];
 }
 
-std::unordered_map<std::string, FiniteElementState> BasePhysics::getCheckpointedStates(int /*cycle*/) const
+std::unordered_map<std::string, FiniteElementState> BasePhysics::getCheckpointedStates(int cycle_to_load) const
 {
-  SLIC_ERROR_ROOT(axom::fmt::format(
-      "loadCheckpointedState and getCheckpointedStates not implemented for physics module {}.", name_));
-  std::unordered_map<std::string, FiniteElementState> empty_container;
-  return empty_container;
+  std::unordered_map<std::string, FiniteElementState> previous_states_map;
+  std::vector<FiniteElementState*>                    previous_states_ptrs;
+
+  if (checkpoint_to_disk_) {
+    for (const auto& state_name : stateNames()) {
+      previous_states_map.emplace(state_name, state(state_name));
+      previous_states_ptrs.emplace_back(const_cast<FiniteElementState*>(&state(state_name)));
+    }
+    StateManager::loadCheckpointedStates(cycle_to_load, previous_states_ptrs);
+    return previous_states_map;
+  } else {
+    for (const auto& state_name : stateNames()) {
+      previous_states_map.emplace(state_name, checkpoint_states_.at(state_name)[static_cast<size_t>(cycle_to_load)]);
+    }
+  }
+
+  return previous_states_map;
 }
 
 double BasePhysics::getCheckpointedTimestep(int cycle) const
 {
   SLIC_ERROR_ROOT_IF(cycle < 0, axom::fmt::format("Negative cycle number requested for physics module {}.", name_));
-  SLIC_ERROR_ROOT_IF(cycle > max_cycle_,
+  SLIC_ERROR_ROOT_IF(cycle > static_cast<int>(timesteps_.size()),
```

Comment:
Thanks for making this error check consistent with the message but should we have a check that we don't go over `max_cycle_`?